ee/disclaim/run_disclaimed_darwin.go

```diff
@@ -0,0 +1,175 @@
+//go:build darwin
+// +build darwin
+
+package disclaim
+
+/*
+#cgo CFLAGS: -mmacosx-version-min=10.14 -Wall -Werror
+#cgo LDFLAGS: -framework Foundation -framework Security -framework System
+
+#include <Security/Authorization.h>
+#include <Security/AuthorizationTags.h>
+#include <spawn.h>
+#include <sys/types.h>
+#include <sys/wait.h>
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+
+int responsibility_spawnattrs_setdisclaim(posix_spawnattr_t attrs, int disclaim);
+
+int spawn_disclaimed(const char *path, char *const argv[], char *const envp[]) {
+    posix_spawnattr_t attrs;
+    int err = posix_spawnattr_init(&attrs);
+    if (err != 0) {
+        return err;
+    }
+
+    // Set the disclaim attribute
+    err = responsibility_spawnattrs_setdisclaim(&attrs, 1);
+    if (err != 0) {
+        posix_spawnattr_destroy(&attrs);
+        return err;
+    }
+
+    pid_t pid;
+    // passing NULL for file_actions, we capture stderr/out from running this as subcommand
+    err = posix_spawn(&pid, path, NULL, &attrs, argv, envp);
+    posix_spawnattr_destroy(&attrs);
+    if (err != 0) {
+        return err;
+    }
+
+    int status;
+    if (waitpid(pid, &status, 0) == -1) {
+        return -1;
+    }
+
+    // WIFEXITED will return non-zero if the child process terminated normally
+    return WIFEXITED(status) ? WEXITSTATUS(status) : -1;
+}
+*/
+import "C"
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os/user"
+	"unsafe"
+
+	"github.com/kolide/launcher/ee/allowedcmd"
+	"github.com/kolide/launcher/pkg/log/multislogger"
+)
+
+type allowedCmdGenerator struct {
+	allowedOpts map[string]struct{}
+	generate    func(ctx context.Context, args []string) (*allowedcmd.TracedCmd, error)
+}
+
+var allowedCmdGenerators = map[string]allowedCmdGenerator{
+	"brew": {
+		allowedOpts: map[string]struct{}{
+			"outdated": {},
+			"--json":   {},
+		},
+		generate: generateBrewCommand,
+	},
+}
+
+func RunDisclaimed(_ *multislogger.MultiSlogger, args []string) error {
```

Comment:
This is effectively a new `main`, and thus a new `ctx` it's going to break traces. I assume there's no way to pass `ctx`, but we might be able to pass whatever we need to connect traces. (either via args, or in the ENV).

I don't think we need to block this PR on this though.